vendor/src/github.com/docker/swarmkit/ca/server.go

```diff
@@ -233,8 +233,9 @@ func (s *Server) IssueNodeCertificate(ctx context.Context, request *api.IssueNod
 					},
 				},
 				Spec: api.NodeSpec{
-					Role:       role,
-					Membership: api.NodeMembershipAccepted,
+					Role:         role,
+					Membership:   api.NodeMembershipAccepted,
+					Availability: request.Availability,
```

Comment:
I don't think we should move `Availability` out of `NodeSpec`.

What I was thinking of was having a wrapper mesasge instead of using `NodeSpec.Availability` directly in `IssueNodeCertificateRequest`. Something like this:

```
message NodeAvailability {
      NodeSpec.Availability availability = 1;
}
```

and in `IssueNodeCertificateRequest`:

```
...
      // Availability allows a user to control the current scheduling status of a node
      NodeAvailability availability = 4;
```

This means `availability` can be nil. If the user does not specify an availability, we can choose a default on the server side.